contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -191,27 +201,34 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
     ) external onlyWhenUnlocked {
         Auction storage auction = validatorAuction[validatorId];
         Validator storage validator = validators[validatorId];
-        require(msg.sender == auction.user, "Only bidder can confirm");
+        require(
+            msg.sender == auction.user ||
+                getValidatorId(msg.sender) == validatorId,
+            "Only bidder or validator can confirm"
+        );
 
         require(
             currentEpoch.sub(auction.startEpoch) % auctionPeriod.add(dynasty) >=
                 auctionPeriod,
             "Confirmation is not allowed before auctionPeriod"
         );
-
+        uint256 perceivedStake = validators[validatorId].amount;
+        address _contract = validators[validatorId].contractAddress;
+        if (_contract != address(0x0)) {
+            perceivedStake = perceivedStake.add(
+                ValidatorShare(_contract).activeAmount()
+            );
+        }
         // validator is last auctioner
-        if (auction.user == NFTContract.ownerOf(validatorId)) {
-            uint256 refund = validator.amount;
-            require(token.transfer(auction.user, refund));
-            validator.amount = auction.amount;
-
+        if (perceivedStake > auction.amount) {
```

Comment:
I'd suggest not allowing dethroning with stake == perceivedStake.

Reply:
yeah makes sense.